x-pack/apm-server/main.go

```diff
@@ -108,6 +109,31 @@ func newTailSamplingProcessor(args beater.ServerParams) (*sampling.Processor, er
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to create Elasticsearch client for tail-sampling")
 	}
+
+	var sampledTracesDataStream sampling.DataStreamConfig
+	if args.Managed {
+		// Data stream and ILM policy are managed by Fleet.
+		sampledTracesDataStream = sampling.DataStreamConfig{
+			Type:      "traces",
+			Dataset:   "sampled",
+			Namespace: args.Namespace,
+		}
+	} else {
+		sampledTracesDataStream = sampling.DataStreamConfig{
+			Type:      "apm",
+			Dataset:   "sampled",
+			Namespace: "traces",
+		}
+		if err := pubsub.SetupDataStream(context.Background(), es,
+			"apm-sampled-traces", // Index template
+			"apm-sampled-traces", // ILM policy
+			"apm-sampled-traces", // Index pattern
```

Comment:
I didn't bother because the feature is subject to breaking changes, and the mapping is very minimal anyway. If we need to version we can always tack on a version later, and this one effectively becomes v0.